addons/web/static/src/webclient/actions/action_service.js

```diff
@@ -1175,6 +1163,18 @@ function makeActionManager(env) {
         // END LEGACY CODE COMPATIBILITY
 
         newController.props = _getViewProps(view, controller.action, controller.views, props);
+
+        let sharedState;
+        if (controller && controller.getSharedState) {
+            sharedState = controller.getSharedState();
+            controller.sharedState = sharedState;
```

Comment:
I don't think it's the correct place to save it. It's shared with all controllers of the action, so it should rather be stored on the action. Obviously we don't want this to leak to the controller instances (via this.props.action), but we already have the problem with the `controllers` key. We'll need to address that anyway.